packages/fxa-settings/src/components/BentoMenu/index.tsx

```diff
@@ -27,18 +27,24 @@ export const BentoMenu = () => {
   const dropDownId = 'drop-down-bento-menu';
   const iconClassNames = 'inline-block w-5 -mb-1 ltr:pr-1 rtl:pl-1';
   const { l10n } = useLocalization();
+  const bentoMenuTitle = l10n.getString(
+    'bento-menu-title',
+    null,
+    'Firefox Bento Menu'
+  );
 
   return (
     <div className="relative self-center flex" ref={bentoMenuInsideRef}>
       <button
         onClick={toggleRevealed}
         data-testid="drop-down-bento-menu-toggle"
-        title={l10n.getString('bento-menu-title', null, 'Firefox Bento Menu')}
+        title={bentoMenuTitle}
+        aria-label={bentoMenuTitle}
         aria-expanded={isRevealed}
-        aria-controls={dropDownId}
+        aria-haspopup={true}
```

Comment:
In most instances, that is the case. That is, in the absence of a value, it implicitly defaults to `true`. However for `aria-haspopup`, it's actually the opposite (and not clearly stated so it does get missed and often assumed otherwise).

In _WAI-ARIA 1.1_, `aria-haspopup` went from `true`/`false` to an enumerated type with six allowed values of varying support (`true`, `menu`, `listbox`, `tree`, `grid`, and `dialog`, where the first two have the widest support and `true` is the same as `menu` - for backward compatibility with _WAI-ARIA 1.0_ - and _should_ be announced as a `menu` pop-up, but isn't always; some screen readers announce that it has a "pop up" but not of type `menu`). [1] The value MUST match the role of the pop-up (again, `true` and `menu` are evaluated to `menu` so the pop-up should have `role="menu"`). [2] Any other value (undefined, false, empty string, etc.) evaluates to `false`.

I'm not clear if `false` is read aloud or ignored. I can't think of a use case for explicitly declaring `false` off the top of my head, and according to the spec, you would think it should just be ignored, yet, after googling, it appears to be read aloud? Or maybe it's acknowledged and explicitly ignored like `aria-hidden` or `role="presentation"` or something similar? (It wasn't read aloud for me on VO, but I might have different settings.)

Footnotes/Code updates:

- [1] I updated the value to explicitly say `menu` instead of `true` because I realized it wasn't consistently being read aloud for me.
- [2] Thank you @LZoog & @pdehaan! I discovered while reviewing your comments yesterday that I had completely forgotten to add `role="menu"` to the pop-up. This could also possibly explain the inconsistent behaviour for [1].
- In addition, I added `!!` to `aria-expanded` as it wasn't consistently being read aloud for me, even though it showed in the DOM and updated as expected (let me know if you have an alternative approach - especially if this might deviate in any way from our style guide).

References:

- [MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-haspopup)
- [w3c discussion on updating spec description](https://github.com/w3c/aria/issues/1209)
- [digitala11y usage and support](https://www.digitala11y.com/aria-haspopup-properties/)
- [WAI-ARIA 1.1 spec](https://www.w3.org/TR/2017/REC-wai-aria-1.1-20171214/#aria-haspopup)

Reply:
Awesome, thanks so much for all your investigation! 😍